Automattic/wp-calypso
# Plugin Scheduler: Check for file mod capabilities

Related to https://github.com/Automattic/dotcom-forge/issues/5874

## Proposed Changes

* Checks the `/v1.2/plugins` API's `file_mod_capabilities` to verify if we can schedule plugins.
* This check applies to both the list and the create pages.
* We disable buttons and display appropriate notices if we can't schedule plugins.
* We start with the assumption that schedule creation is possible, and wait for the API to return to show disable states if necessary. This seemed to be the safest thing to do.

## Testing Instructions



* Apply this PR: https://github.com/Automattic/jetpack/pull/36238 and this diff: D141065-code by following instructions there.
* Go to `/plugins/scheduled-updates/site` 
* Go to `/plugins/scheduled-updates/create/site` 
* Go to `/plugins/scheduled-updates/edit/site/id` 
* You should see the default state.

Now, to test a bad result, add a line here: https://github.com/Automattic/jetpack/pull/36238/files#diff-16bb61901772bfecbc9020e4b205240d10a347c6f5e6ac61fe795186bcc4d997R234: 

```
		$reasons_can_not_modify_files['is_sub_site'] = __( 'The site is not the main network site', 'jetpack-scheduled-updates' );

```

* Repeat the above checks to see the new UI.

![CleanShot 2024-03-05 at 14 56 09](https://github.com/Automattic/wp-calypso/assets/533/cb22e0eb-5621-4433-a565-10fd3ca85c31)
![CleanShot 2024-03-05 at 14 59 12](https://github.com/Automattic/wp-calypso/assets/533/90975500-d896-4efa-8a0b-d5d219108f62)
![CleanShot 2024-03-07 at 13 34 52](https://github.com/Automattic/wp-calypso/assets/533/e18b9b5c-e14c-4488-800b-07bd04cc030e)



## Pre-merge Checklist
